platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronizedInstallerImpl.java

```diff
@@ -0,0 +1,598 @@
+        () -> {
+          LOGGER.trace(
+              "Waiting for service with pid [{}] to reflect configuration updates...", servicePid);
+          return listener.isUpdated();
+        };
+
+    LOGGER.info("Updating configuration of service with pid [{}].", servicePid);
+
+    try {
+      Configuration config = configAdmin.getConfiguration(servicePid, bundleLocation);
+      config.setBundleLocation(bundleLocation);
+      config.update(toDic(properties));
+
+      LOGGER.debug("Updated configuration of service with pid [{}].", servicePid);
+      wait(
+          isServiceUpdated,
+          getRemainingTime(startTime, maxWaitTime),
+          DEFAULT_POLLING_INTERVAL,
+          "Managed service failed to update within [" + maxWaitTime + "] ms.");
+    } catch (InterruptedException | SynchronizedInstallerException e) {
+      throw e;
+    } catch (IOException e) {
+      throw new SynchronizedInstallerException(
+          "Failed to update managed service configuration with pid of [" + servicePid + "]", e);
+    } catch (Exception e) {
+      throw new SynchronizedInstallerException(
+          "Exception was thrown while waiting for service with pid of ["
+              + servicePid
+              + "] to be updated.",
+          e);
+    } finally {
+      registration.unregister();
+    }
+  }
+
+  @Override
+  public void waitForServiceToBeAvailable(String servicePid) throws InterruptedException {
+    waitForServiceToBeAvailable(DEFAULT_MAX_SERVICE_WAIT, servicePid);
+  }
+
+  @Override
+  public void waitForServiceToBeAvailable(long maxWaitTime, String servicePid)
+      throws InterruptedException {
+    ServiceTracker st = getServiceTracker(bundleContext, getServiceIdFilter(servicePid));
+    st.open(true);
+
+    Callable<Boolean> isServiceAvailable =
+        () -> {
+          LOGGER.trace("Waiting for service with pid [{}] to be registered", servicePid);
+          return st.getService() != null;
+        };
+
+    try {
+      wait(
+          isServiceAvailable,
+          maxWaitTime,
+          DEFAULT_POLLING_INTERVAL,
+          "Managed service failed to appear after [" + maxWaitTime + "] ms.");
+    } finally {
+      st.close();
+    }
+  }
+
+  @Override
+  public void installFeatures(String feature, String... additionalFeatures)
+      throws InterruptedException {
+    installFeatures(EnumSet.noneOf(FeaturesService.Option.class), feature, additionalFeatures);
+  }
+
+  @Override
+  public void installFeatures(
+      EnumSet<FeaturesService.Option> options, String feature, String... additionalFeatures)
+      throws InterruptedException {
+    installFeatures(DEFAULT_MAX_FEATURE_WAIT, options, feature, additionalFeatures);
+  }
+
+  @Override
+  public void installFeatures(
+      long maxWaitTime,
+      EnumSet<FeaturesService.Option> options,
+      String feature,
+      String... additionalFeatures)
+      throws InterruptedException {
+    Set<String> featuresToInstall =
```

Comment:
❓ What happens if the resulting set is empty?

Reply:
Added a check for this as well.